src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -17,6 +17,42 @@ namespace Microsoft.Web.LibraryManager.Contracts
     /// </summary>
     public static class FileHelpers
     {
+        /// <summary>
+        /// Writes the stream to a temporary file first, then moves the temporary file to the destination file
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="sourceStream"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public static async Task<bool> SafeWriteToFileAsync(string fileName, Stream sourceStream, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string tempFileName = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.InternetCache),
+                "libman",
+                fileName.GetHashCode().ToString(),
```

Comment:
>GetHashCode() [](start = 25, length = 13)

Does string.GetHashCode give enough entropy to avoid conflicts?  Is it case-aware?

Reply:
I believe it's case aware, so we should probably do ToLower on the filename. I'm using library file name in addition to the full path hash, so I believe it should be enough. Theoretically two different full paths could get hashed to the same value, in which case the file names will be different. If we want to be super-safe, I can check for an existing folder I guess. Also since we are not doing resume for now, we could just get unique temp path to eliminate the possibility of hashing to the same value. What do you think?

Reply:
Changed to Path.GetTemporaryFiles and added a fallback to the old behavior in case if that's not implemented on some platform.